test/built-ins/Object/getOwnPropertyDescriptors/proxy-undefined-descriptor.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2016 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Object.getOwnPropertyDescriptors should perform observable operations in the correct order
+esid: sec-object.getownpropertydescriptors
+author: Jordan Harband
+features: [Proxy]
+includes: [proxyTrapsHelper.js]
+---*/
+
+var key = "a";
+var ownKeys = [key];
+var badProxyHandlers = allowProxyTraps({
+  getOwnPropertyDescriptor: function () {},
+  ownKeys: function () {
+	return ownKeys;
+  }
+});
+var proxy = new Proxy({}, badProxyHandlers);
+
+var keys = Object.keys(proxy);
+assert.sameValue(keys, ownKeys, 'Object.keys returns the result of the [[OwnKeys]] trap');
```

Comment:
When it comes to a "deepEquals" helper, we have had arguments both for and against--see gh-149. For my part, I think that a complete "deepEquals" is generally much more complicated than what we need for Test262 tests. [`harness/compareArray.js`](https://github.com/tc39/test262/blob/fce8b5852d7d9d3839dea6922bae306934ac115a/harness/compareArray.js) is available for shallow comparison of array-like objects, though even that is generally more abstraction than I'm comfortable with.

For instance, this test compares two arrays of length 1, so it can be written with two assertions (no need for external functions, loops, or recursion of any kind):

```js
assert.sameValue(keys.length, 1);
assert.sameValue(keys[0], 'a');
```

Though to Caitlin's point, it's probably a good idea to also add in `assert.notSameValue(keys, ownKeys);`.

Beyond just being spartan for the sake of it, explicit tests like these are less prone to false positives resulting from comparing one computed value against another ([example here](https://github.com/tc39/test262/pull/602#discussion_r61342819))